projects/iio_demo/src/main.c

```diff
@@ -0,0 +1,203 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#ifdef XILINX_PLATFORM
+#include <xparameters.h>
+#include <xil_cache.h>
+#include <xil_cache.h>
+#endif // XILINX_PLATFORM
+#include "app_config.h"
+#include "parameters.h"
+#include "error.h"
+#include "iio_app.h"
+#include "iio_demo.h"
+#include "irq.h"
+#include "irq_extra.h"
+#include "uart.h"
+#include "uart_extra.h"
+
+
+static struct uart_desc *uart_desc;
+
+/**
+ * iio_uart_write() - Write data to UART device wrapper.
+ * @buf - Pointer to buffer containing data.
+ * @len - Number of bytes to write.
+ * @Return: SUCCESS in case of success, FAILURE otherwise.
+ */
+static ssize_t iio_uart_write(const char *buf, size_t len)
+{
+	return uart_write(uart_desc, (const uint8_t *)buf, len);
+}
+
+/**
+ * iio_uart_read() - Read data from UART device wrapper.
+ * @buf - Pointer to buffer containing data.
+ * @len - Number of bytes to read.
+ * @Return: SUCCESS in case of success, FAILURE otherwise.
+ */
+static ssize_t iio_uart_read(char *buf, size_t len)
+{
+	return uart_read(uart_desc, (uint8_t *)buf, len);
+}
+
+/***************************************************************************//**
+ * @brief main
+*******************************************************************************/
+int main(void)
+{
+	int32_t status;
+
+	/**
+	 * iio application configurations.
+	 */
+	struct iio_app_init_param iio_app_init_par;
+
+	/**
+	 * iio demo configurations.
+	 */
+	struct iio_demo_init_param iio_demo_init_par;
+
+	/**
+	 * UART server read/write callbacks.
+	 */
+	struct iio_server_ops uart_iio_server_ops;
+
+	/**
+	 * iio application instance descriptor.
+	 */
+	struct iio_app_desc *iio_app_desc;
+
+	/**
+	 * iio instance descriptor.
+	 */
+	struct iio_demo_desc *iio_demo_desc;
+
+#ifdef XILINX_PLATFORM
+	/**
+	 * Xilinx platform dependent initialization for IRQ.
+	 */
+	struct xil_irq_init_param xil_irq_init_par = {
+		.type = IRQ_PS,
+	};
+#endif // XILINX_PLATFORM
+
+	/**
+	 * IRQ initial configuration.
+	 */
+	struct irq_init_param irq_init_param = {
+		.irq_ctrl_id = INTC_DEVICE_ID,
+#ifdef XILINX_PLATFORM
+		.extra = &xil_irq_init_par,
+#endif // XILINX_PLATFORM
+	};
+
+
+	/**
+	 * IRQ instance.
+	 */
+	struct irq_ctrl_desc *irq_desc;
+
+	/**
+	 * Initialization for UART.
+	 */
+	struct uart_init_param uart_init_par;
+
+#ifdef XILINX_PLATFORM
+	/**
+	 * Xilinx platform dependent initialization for UART.
+	 */
+	struct xil_uart_init_param xil_uart_init_par;
```

Comment:
I think this can be moved before line 153. This way irq initialization will be in one place and uart in another, not mixed.